beacon-chain/blockchain/receive_block.go

```diff
@@ -468,6 +468,9 @@ func (s *Service) validateStateTransition(ctx context.Context, preState state.Be
 	stateTransitionStartTime := time.Now()
 	postState, err := transition.ExecuteStateTransition(ctx, preState, signed)
 	if err != nil {
+		if ctx.Err() == context.DeadlineExceeded {
```

Comment:
Why not handle this for all context error? is deadline the only one? what about timeout?